### PR TITLE
parser: support username without quotes

### LINKED ISSUE #3693: create user without quotes failed
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

exec sql `create user user1@localhost;`

2. What did you expect to see?

```
mysql> create user user1@localhost;
Query OK, 0 rows affected (0.00 sec)
```

3. What did you see instead?

```
mysql> create user uesr1@localhost;
ERROR 1105 (HY000): line 0 column 17 near "@localhost" (total length 27)
```

but add quotes will succeed:

```
create user 'uesr1'@'localhost';
```

4. What version of TiDB are you using (`tidb-server -V`)?

```
Git Commit Hash: d1328e1b7282c026e579a413db3ad386c15fdabf
UTC Build Time:  2017-06-22 03:37:28
```

### PR DESCRIPTION
**Pull request by breeswish**
Fix #3693

### CODE REVIEW

**Comment by breeswish** [comment]
@zimulala @tiancaiamao PTAL

**Comment by zimulala** [comment]
PTAL @winkyao @jackysp

File: parser/parser_test.go [hunk 1]
```diff
@@ -1412,6 +1412,24 @@ func (s *testParserSuite) TestPrivilege(c *C) {
...
+		{`CREATE USER test`, true},
+		{"CREATE USER `test`", true},
+		{"CREATE USER test-user", false},
+		{"CREATE USER test.user", false},
+		{"CREATE USER 'test-user'", true},
+		{"CREATE USER `test-user`", true},
+		{"CREATE USER test.user", false},
+		{"CREATE USER 'test.user'", true},
+		{"CREATE USER `test.user`", true},
+		{"CREATE USER uesr1@localhost", true},
```
Comment (breeswish) [inline parser/parser_test.go]: It is copied from the original issue https://github.com/pingcap/tidb/issues/3693

File: parser/parser.y [hunk 2]
```diff
@@ -47,6 +47,8 @@ import (
 %token	<ident>
 	/*yy:token "%c"     */	identifier      "identifier"
 	/*yy:token "\"%c\"" */	stringLit       "string literal"
+	singleAtIdentifier		"identifier with single leading at"
```
Comment (breeswish) [inline parser/parser.y]: It's some kind of workaround for the current lexer. the lexer will convert `user@host` into token `user` and `@host` while for `"user"@"host"` is `user`, `@`, `host`. differentiating `@host` and `@userVariable` in the lexer may require knowledge about the current parsing context, which is lacking now.

**Comment by XuHuaiyu** [comment]
please resolve the conflicts

**Comment by shenli** [comment]
LGTM
Good job!

**Pull request merged**